cmd/jaeger/internal/extension/remotesampling/config.go

```diff
@@ -104,6 +107,10 @@ func (cfg *Config) Validate() error {
 		return errNegativeInterval
 	}
 
+	if cfg.File != nil && (cfg.File.DefaultSamplingProbability > 1 || cfg.File.DefaultSamplingProbability < 0) {
```

Comment:
I think you should be able to achieve similar validation by adding a `valid` tag to the field, perhaps `valid:range(0, 1)`